### PR TITLE
Implement public TableClient ctor that takes Uri + creds

### LINKED ISSUE #13729: Implement public TableClient ctor that takes Uri + Credentials
Allows developers to start directly with a TableClient instance if we've got a URI to one.

### PR DESCRIPTION
**Pull request by christothes**
closes #13729

### CODE REVIEW

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs [hunk 1]
```diff
@@ -28,6 +28,86 @@ public class TableClient
...
+        /// </summary>
+        /// <param name="tableName">The name of the table with which this client instance will interact.</param>
+        /// <param name="endpoint">
+        /// A <see cref="Uri"/> referencing the table service.
+        /// This is likely to be similar to "https://{account_name}.table.core.windows.net/" or "https://{account_name}.table.cosmos.azure.com/".
+        /// </param>
+        /// <param name="options">
+        /// Optional client options that define the transport pipeline policies for authentication, retries, etc., that are applied to every request.
+        /// </param>
+        public TableClient(string tableName, Uri endpoint, TableClientOptions options = null)
```
Comment (meanjeekim) [inline sdk/tables/Azure.Data.Tables/src/TableClient.cs:42]: Just wondering what the thinking was behind a ctor that doesn't take in `TableClientOptions` versus taking in one that defaults!

File: sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs [hunk 2]
```diff
@@ -22,10 +22,27 @@ public class TableServiceClient
         private readonly string _version;
         internal readonly bool _isPremiumEndpoint;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TableServiceClient"/>.
+        /// </summary>
+        /// <param name="endpoint">
+        /// A <see cref="Uri"/> referencing the table service.
```
Comment (meanjeekim) [inline sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs:29]: `TableServiceClient` will always refer to one account, right?

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs [hunk 3]
```diff
@@ -28,6 +28,86 @@ public class TableClient
...
+        /// <param name="tableName">The name of the table with which this client instance will interact.</param>
+        /// <param name="endpoint">
+        /// A <see cref="Uri"/> referencing the table service.
+        /// This is likely to be similar to "https://{account_name}.table.core.windows.net/" or "https://{account_name}.table.cosmos.azure.com/".
+        /// </param>
+        /// <param name="credential">The shared key credential used to sign requests.</param>
+        public TableClient(string tableName, Uri endpoint, TableSharedKeyCredential credential)
+            : this(tableName, endpoint, new TableSharedKeyPipelinePolicy(credential), null)
+        {
+            Argument.AssertNotNull(tableName, nameof(tableName));
```
Comment (meanjeekim) [inline sdk/tables/Azure.Data.Tables/src/TableClient.cs:65]: Should we always have an xml comment for thrown exceptions? Not sure what the policy is on that.

**Review by meanjeekim — APPROVED** [review]
I just left some questions that I had! :D

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs (hunk 3 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableClient.cs:65]: yes, good catch!

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs (hunk 1 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableClient.cs:42]: It's specified in the [guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-client-ctor)

**Pull request merged**